### PR TITLE
Add contextual music rarity

### LINKED ISSUE #5552: Make context specific music tracks more likely to play in the microbe stage
I think a good way would be to prevent most of the normal tracks from playing when there's an active context to make it like a 30% chance at least to play the context specific tracks (that way they aren't repetitive but still most players would probably hear them rather than being like a 1/7 chance only). I *think* the jukebox system can specify to not play a track if a context is active, but if that part wasn't fully implemented a tiny bit of C# programming will be needed for this.

### PR DESCRIPTION
**Pull request by CI09**
**Brief Description of What This PR Does**

Adds rarity constant and math for it

**Related Issues**

Closes #5552

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [x] Functionality is confirmed working by another person (see above checklist link)
- [x] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

File: src/general/Jukebox.cs [hunk 1]
```diff
@@ -556,6 +556,18 @@ private void PlayNextTrackFromList(TrackList list, Func<int, AudioPlayer> getPla
...
             return;
 
+        var random = new XoShiRo128starstar();
+
+        var contextMusicOnly = tracks.Where(c => c.ExclusiveToContexts != null).ToArray();
+
+        if (contextMusicOnly.Length > 0)
+        {
+            if (random.Next(0, Constants.CONTEXTUAL_MUSIC_RARITY) == 0)
+            {
```
Comment (dligr) [inline src/general/Jukebox.cs:566]: Would it be better to make the random check before getting the context tracks for some extra performance?

Comment (CI09) [reply src/general/Jukebox.cs:566]: Smart

**Review by dligr — APPROVED** [review]
I have tested this to work in game. Code changes make sense to me too.

**Review by hhyyrylainen — APPROVED** [review]
I improved the memory handling and code clarity here. I also added a TODO comment about fixing playing the same track over and over with this new code: https://github.com/Revolutionary-Games/Thrive/issues/5615 

I'll merge this now and assume my changes didn't break the actual functionality here.

**Pull request merged**